### ISSUE TITLE
Wrong error message for `this` in function

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/TypeScript#26447]
**TypeScript Version:**  3.1.0-dev.20180810

**Code**

```ts
function f() {
    this.x;
}
```

**Expected behavior:**

Error that `this` is implicitly `any`.

**Actual behavior:**

```
src/a.ts:2:5 - error TS7041: The containing arrow function captures the global value of 'this' which implicitly has type 'any'.
```

...what arrow function?
(Ref: #25978)

**Comment by andy-ms** [comment]
On that note, we should also have a good error message in the following situation:
```ts
class C {
    m() {
        this.m();
        function f() {
            this.m();
        }
    }
}
```
This is a common error and it's not obvious why `this` works in one place but not the other. The error message should indicate the place where `this` was shadowed.

**Comment by nermin99** [comment]
How do you disable this rule?
